gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireContainer.java

```diff
@@ -475,6 +475,17 @@ public GemFireContainer(ContainerKey id, Properties properties)
     initTableFlags();
   }
 
+  private ExternalCatalog waitForHiveCatalogInit() {
+    ExternalCatalog ret;
+    int cnt = -1;
+    // Retrying after sleep of some millisecs to reduce the worst case
+    // of delaying that put for a large period of time
+    while((ret = Misc.getMemStore().getExternalCatalog()) == null && cnt <= 20) {
```

Comment:
20 tries would be too small no? The retry sleep is 10ms (and 200ms on 20th invocation) so total 400ms. Maybe more like 5000. GfxdConstants.HA_NUM_RETRIES is 10000 which is used elsewhere so can use that.

Reply:
Was using this GfxdConstants.HA_NUM_RETRIES earlier. Will reinstate.